beetsplug/rewrite.py

```diff
@@ -34,9 +34,7 @@ def rewriter(field, rules):
     def fieldfunc(item):
         value = item._values_fixed[field]
         for pattern, replacement in rules:
-            if pattern.match(value.lower()):
-                # Rewrite activated.
-                return replacement
+            return re.sub(pattern, replacement, value)
```

Comment:
Thanks for thinking this through! I think this reasoning is on the right track.

I'm a little worried about testing for subpattern references, though, since I can imagine people wanting to write plain replacement strings (e.g., `\s+EP$` with the empty string to strip off the redundant ends of EP titles) if we open up arbitrary find/replace rules. So I'm not sure that `\1` occurrences are a perfect litmus test.

Perhaps we just need a plugin-wide option to switch between the old and the new behavior?